File: level_1/problems/reverse_string.py
from typing import List


class Solution:
    def reverseString(self, s: List[str]) -> None:
        stack = []

        for n in s:
            stack.append(n)

        for i in range(len(s)):
            s[i] = stack.pop()

        print(f"stack values: {stack}")
        print(f"vector values: {s}")


s = ["h", "e", "l", "l", "o"]
sol = Solution()
print(sol.reverseString(s))

File: level_1/problems/reverse_string.cpp
#include <iostream>
#include <vector>

using namespace std;
class Solution {
    public:
        void reverseString(vector<char>& s) {
            char temp;
            int l = 0;
            int r = s.size() - 1;

            while (l < r) {
                temp = s[l];
                s[l] = s[r];
                s[r] = temp;
                l++;
                r--;
            }
            for (const char& c : s) {
                cout << c << " ";
            }
            cout << endl;
        }
};

int main() {
    Solution sol;
    vector<char> s = {'h', 'e', 'l', 'l', 'o'};

    sol.reverseString(s);

    return 0;
}
